lib/interactive_viewer.ml

```diff
@@ -115,44 +115,163 @@ let toggle_view_mode () =
   | Normal -> Lwd.set view_mode SideBySide
   | SideBySide -> Lwd.set view_mode Normal
 
-let rec split_and_align_hunk hunks mine_acc their_acc =
+(* Somewhere here we are not organizing *)
+(* Adding a type for changes / common / empty *)
+(* Rewrite this function to tag properly  *)
+
+type line = Change of change_origin * string | Common of string | Empty
+and change_origin = Mine | Their
+
+let rec split_and_align_hunk hunks mine_acc their_acc mine_hunk_length
+    their_hunk_length =
+  let rec buffer_changes temp_mine temp_their mine_hunk_length their_hunk_length
+      hunks =
+    match hunks with
+    | [] -> (temp_mine, temp_their, mine_hunk_length, their_hunk_length, hunks)
+    | `Common _ :: _ ->
+        ( temp_mine,
+          temp_their,
+          mine_hunk_length + 1,
+          their_hunk_length + 1,
+          hunks )
+    | `Mine line :: t ->
+        buffer_changes
+          (Change (Mine, line) :: temp_mine)
+          temp_their (mine_hunk_length + 1) their_hunk_length t
+    | `Their line :: t ->
+        buffer_changes temp_mine
+          (Change (Their, line) :: temp_their)
+          mine_hunk_length (their_hunk_length + 1) t
+  in
+
+  let append_balanced temp_mine temp_their mine_hunk_length their_hunk_length =
```

Comment:
This function is returning lists in the right direction while the rest of the code works with reversed lists. If this inconsistency is important, then it should be documented.

Reply:
The documentation helps but I'm not sure the reversal helps. Reversing the list here is an extra step and handling the reversed-list elsewhere has the potential to mix up the orders.
Why not working with reversed lists like the other functions ?